src/main/java/commonuseractions/CommonActionMethods.java

```diff
@@ -630,7 +649,174 @@ public String currentDate(int plusdays, String format) {
+		} catch (Exception e) {
+			flag = false;
+		}
+		return flag;
+	}
+	/**
+	 * @method To swipe down until the element appears, If need to click after Swipe set third parameter as True.
+	 * @param element
+	 * @param name
+	 * @param click
+	 * @throws Exception
+	 */
+	public void swipeDown(WebElement element, String name, boolean click) throws Exception {
+		int scrollPoints = 0;
+		while (true) {
+			System.out.println(element);
+			if (isElementPresent(element) == false) {
+				 
+				System.out.println(windowSize);
+				PointerInput finger = new PointerInput(PointerInput.Kind.TOUCH,
+						"finger");
+				Sequence swipeDown = new Sequence(finger, 1);
+				swipeDown.addAction(finger.createPointerMove(Duration.ZERO, Origin.viewport(), windowSize.width / 2,
+						windowSize.height / 2)).addAction(finger.createPointerDown(MouseButton.LEFT.asArg()))
+						.addAction(finger.createPointerMove(Duration.ofMillis(700), Origin.viewport(),
+								windowSize.height / 2, windowSize.height / 2 + windowSize.height / 2))
+						.addAction(finger.createPointerUp(MouseButton.LEFT.asArg()));
+				appiumdriver.perform(Arrays.asList(swipeDown));
+				logMessage(" Element not in view, Scrolling up ");
+				scrollPoints++;
+				if(scrollPoints>10) {
+					logErrorMessage(" Element not found ");
+					break;
+				}
+			} else if (isElementPresent(element) == true) {
+				if (click) {
+					clickMethod(element, name);
+				}
+				logMessage("Element is in view ");
+				break;
+			}
+		}
+	}
+	/**
+	 * @method To swipe down until the element appears and sends text to the input field.
+	 * @param element
+	 * @param name
+	 * @param click
+	 * @throws Exception
+	 */
+	public void swipeDown(WebElement element, String name, boolean sendkey,String keytoSend) throws Exception {
+		int scrollPoints = 0;
+		while (true) {
+			System.out.println(element);
+			if (isElementPresent(element) == false) {
+				 
+				System.out.println(windowSize);
+				PointerInput finger = new PointerInput(PointerInput.Kind.TOUCH,
+						"finger");
+				Sequence swipeDown = new Sequence(finger, 1);
+				swipeDown.addAction(finger.createPointerMove(Duration.ZERO, Origin.viewport(), windowSize.width / 2,
+						windowSize.height / 2)).addAction(finger.createPointerDown(MouseButton.LEFT.asArg()))
+						.addAction(finger.createPointerMove(Duration.ofMillis(700), Origin.viewport(),
+								windowSize.height / 2, windowSize.height / 2 + windowSize.height / 2))
+						.addAction(finger.createPointerUp(MouseButton.LEFT.asArg()));
+				appiumdriver.perform(Arrays.asList(swipeDown));
+				logMessage(" Element not in view, Scrolling up ");
+				scrollPoints++;
+				if(scrollPoints>10) {
+					logErrorMessage(" Element not found ");
+					break;
+				}
+			} else if (isElementPresent(element) == true) {
+				if (sendkey) {
+					sendKeysMethod(element, keytoSend);
+				}
+				logMessage("Element is in view ");
+				break;
+			}
+		}

```

Comment:
Please remove lines from 655 to 820 then take a pull from develop which has these refactored methods for swipe actions.